src/server/system_services/schemas/master_key_schema.js

```diff
@@ -0,0 +1,34 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+module.exports = {
+    id: 'master_key_schema',
+    type: 'object',
+    required: [
+        '_id',
+    ],
+    properties: {
+        _id: { objectid: true },
+        description: { type: 'string' },
+        // If missing - encrypted with root key
+        // 1. ENV - Mounted key from Kubernetes secret
+        // 2. HSM configuration TBD - Get key from Vault
+        master_key_id: {
```

Comment:
@guymguym 
Can you please take a look at the oneOf, not sure if we shall push it like that?
Maybe use a custom ObjectId for the master_key that will be constant instead of the string 'noobaa-root-master-key'

Reply:
The reason I'm not a fan of oneOf is that it defines the property with a confusing type, and unless we really feel like this is the only option we can do it (like in our bigint type for example) but if it's easy to separate to different properties that each has a concise type I think it should be preferred.

Reply:
it's object_id or the root_key name - so it's very easy to separate between those two, I agree that it's better than hard coded objectId